app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js

```diff
@@ -60,40 +60,28 @@ define([
          */
         _updateWishlistData: function (event) {
             var dataToAdd = {},
-                isFileUploaded = false,
-                self = this;
+                self = this,
+                element = $(event.target);
 
-            if (event.handleObj.selector == this.options.qtyInfo) { //eslint-disable-line eqeqeq
-                this._updateAddToWishlistButton({});
-                event.stopPropagation();
-
-                return;
+            if (element.is('input[type=file]') && element.val()) {
+                this.bindFormSubmit();
             }
-            $(event.handleObj.selector).each(function (index, element) {
-                if ($(element).is('input[type=text]') ||
-                    $(element).is('input[type=email]') ||
-                    $(element).is('input[type=number]') ||
-                    $(element).is('input[type=hidden]') ||
-                    $(element).is('input[type=checkbox]:checked') ||
-                    $(element).is('input[type=radio]:checked') ||
-                    $(element).is('textarea') ||
-                    $('#' + element.id + ' option:selected').length
-                ) {
-                    if ($(element).data('selector') || $(element).attr('name')) {
-                        dataToAdd = $.extend({}, dataToAdd, self._getElementData(element));
-                    }
-
-                    return;
-                }
 
-                if ($(element).is('input[type=file]') && $(element).val()) {
-                    isFileUploaded = true;
+            if (!element.is(this.options.qtyInfo) &&
+                element.is('input[type=text]') ||
+                element.is('input[type=email]') ||
+                element.is('input[type=number]') ||
+                element.is('input[type=hidden]') ||
+                element.is('input[type=checkbox]:checked') ||
+                element.is('input[type=radio]:checked') ||
+                element.is('textarea') ||
+                $('#' + element.id + ' option:selected').length
+            ) {
+                if (element.data('selector') || element.attr('name')) {
+                    dataToAdd = $.extend(dataToAdd, self._getElementData(element));
```

Comment:
If this `.each` is really unnecessary here then I believe `self` can also be removed in favour of using `this` directly.